sub-packages/bionemo-geneformer/src/bionemo/geneformer/scripts/geneformer_mlm_loss_eval.py

```diff
@@ -0,0 +1,295 @@
+        self.my_token_dict = deepcopy(my_token_dict)
+        self.nv_tokenizer = deepcopy(nv_tokenizer)
+        self.n_tokens_nv = len(self.nv_tokenizer.vocab)
+        self.n_tokens_hf = len(my_token_dict)
+
+        # nvidia tokenizer has [cls] and [pad] first along with some others that do not overlap. This mapper
+        hf_ordered_nv_tokenizer = {
+            self.nv_tokenizer.pad_token: my_token_dict["<pad>"],
+            self.nv_tokenizer.mask_token: my_token_dict["<mask>"],
+            self.nv_tokenizer.cls_token: my_token_dict["<cls>"],
+            self.nv_tokenizer.sep_token: my_token_dict["<eos>"],  # name doesn't really matter here
+        }
+        tokens = list(my_token_dict.items())
+        for k, t in tokens[:4]:
+            assert k.startswith("<")
+
+        missing_nv_tokens = []
+        extra_tokens_not_covered = []
+        for ens, idx in list(my_token_dict.items())[4:]:
+            assert ens.startswith("ENSG")
+            if ens in nv_tokenizer.vocab.keys():
+                hf_ordered_nv_tokenizer[ens] = idx
+            else:
+                if idx < self.n_tokens_hf:
+                    missing_nv_tokens.append(idx)
+                else:
+                    extra_tokens_not_covered.append(idx)
+        self.hf_ordered_nv_tokenizer = hf_ordered_nv_tokenizer
+        self.extra_tokens_not_covered = extra_tokens_not_covered
+        self.register_buffer("missing_nv_tokens", torch.tensor(missing_nv_tokens, dtype=int))
+
+    @property
+    def device(self) -> torch.device:
+        """Return the device of this model."""
+        # This is populated through the self.register_buffer call in init.
+        return self.missing_nv_tokens.device
+
+    def get_tokenizer(self) -> GeneTokenizer:
+        """Return the filtered tokenizer with keys that match the order of the nv model."""
+        nv_tok = deepcopy(self.nv_tokenizer)
+        # HF tokenizer only has pad and mask, no other special tokens.
+        nv_tok.special_tokens = (nv_tok.mask_token, nv_tok.pad_token)  # type: ignore
+        nv_tok.vocab = self.hf_ordered_nv_tokenizer
+        nv_tok.decode_vocab = {v: k for k, v in nv_tok.vocab.items()}
+        return nv_tok
+
+    def forward(self, *args, **kwargs):
+        """Run forward and return the logits."""
+        logits = self.model(*args, **kwargs).logits
+        # logits[:, :, self.missing_nv_tokens] = -torch.inf
+        # breakpoint()
+        return logits
+
+
+def main(
+    model_path: Path | None,
+    hf_model_path: str,
+    dataset_path: Path,
+    token_dictionary_path: Path,
+    mask_prob: float = 0.15,
+    batch_size: int = 16,
+    precision: str = "bf16-mixed",
+    config_class: Type[BioBertConfig] = GeneformerConfig,
+    seed: int = 513,
+):
+    """Inference function (requires DDP and only training data that fits in memory)."""
+    # This is just used to get the tokenizer :(
+    train_data_path: Path = (
```

Comment:
Why do we only want this training data path?

Reply:
This has the tokenizer in it.

Reply:
Would this work with another tokenizer? If not, I can see the training path download + test case issue. But in that case, couldn't we have a test case that also downloads this data from PBSS to that location?

Reply:
This script is very one off for this purpose. I was not trying to make something too general. We can update it in the future to make it compatible with a future version of geneformer that uses a different tokenizer. For now though my goal was to port this script over from bionemo1 and update the numbers in the modelcard.

Reply:
Created an issue for this #402